sos/plugins/maas.py

```diff
@@ -53,18 +53,42 @@ def _remote_api_login(self):
         return ret['status'] == 0
 
     def setup(self):
-        self.add_copy_spec([
-            "/etc/squid-deb-proxy",
-            "/etc/maas",
-            "/var/lib/maas/dhcp*",
-            "/var/log/apache2*",
-            "/var/log/maas*",
-            "/var/log/upstart/maas-*",
-        ])
-        self.add_cmd_output([
-            "apt-cache policy maas-*",
-            "apt-cache policy python-django-*",
-        ])
+        snap_list_output = self.exec_cmd('snap list maas')
```

Comment:
I think it deserves some comment why the plugin behaves so differently based on the outcome of this command.

Reply:
Further, as this "if" statement changes the plugin behaviour so heavily, does not it make sense to have it e.g. a child/parent class for it? It sounds strange if the same plugin is expected to do quite different stuff during different executions..? (I am OK with the current code, just.. won't a child plugin be a better option?)

Reply:
Good point, have added some commentary.